wakuv2/common/envelope.go

```diff
@@ -0,0 +1,79 @@
+package common
+
+import (
+	"encoding/json"
+
+	"github.com/ethereum/go-ethereum/common/hexutil"
+	"github.com/waku-org/go-waku/waku/v2/protocol/pb"
+)
+
+// Envelope contains information about the pubsub topic of a WakuMessage
+// and a hash used to identify a message based on the bytes of a WakuMessage
+// protobuffer
+type Envelope interface {
+	Message() *pb.WakuMessage
+	PubsubTopic() string
+	Hash() pb.MessageHash
+}
+
+type envelopeImpl struct {
+	msg   *pb.WakuMessage
+	topic string
+	hash  pb.MessageHash
+}
+
+type tmpWakuMessageJson struct {
+	Payload        []byte  `json:"payload,omitempty"`
+	ContentTopic   string  `json:"contentTopic,omitempty"`
+	Version        *uint32 `json:"version,omitempty"`
+	Timestamp      *int64  `json:"timestamp,omitempty"`
+	Meta           []byte  `json:"meta,omitempty"`
+	Ephemeral      *bool   `json:"ephemeral,omitempty"`
+	RateLimitProof []byte  `json:"proof,omitempty"`
+}
+
+type tmpEnvelopeStruct struct {
```

Comment:
- `tmp` is confusing
- Not need for `Struct` suffix
```suggestion
type NWakuEnvelope struct {
```

Reply:
Though perhaps I got the idea wrong?

Reply:
Notice that those couple structs are not exported and we just use them in `NewEnvelope` to unmarshal a JSON and then use the results to create an envelope

Reply:
I see that, yet. But even for private data types, I believe it's better to keep the intentions clear.